sdk/monitor/Azure.Monitor.Query/src/LogsQueryClient.cs

```diff
@@ -487,7 +464,80 @@ internal static QueryBody CreateQueryBody(string query, MonitorQueryTimeRange ti
             return queryBody;
         }
 
-        private async Task<Response<LogsQueryResult>> ExecuteAsync(string workspaceId, string query, MonitorQueryTimeRange timeRange, LogsQueryOptions options, bool async, CancellationToken cancellationToken = default)
+        private async Task<Response<LogsBatchQueryResultCollection>> ExecuteBatchAsync(LogsBatchQuery batch, bool async, CancellationToken cancellationToken = default)
+        {
+            Response<LogsBatchQueryResultCollection> ConvertBatchResponse(BatchResponse response, Response rawResponse)
+            {
+                List<LogsBatchQueryResult> batchResponses = new List<LogsBatchQueryResult>();
+                foreach (var innerResponse in response.Responses)
+                {
+                    var body = innerResponse.Body;
+                    body.Status = innerResponse.Status switch
+                    {
+                        >= 400 => LogsQueryResultStatus.Failure,
+                        _ when body.Error != null => LogsQueryResultStatus.PartialFailure,
+                        _ => LogsQueryResultStatus.Success
+                    };
+                    body.Id = innerResponse.Id;
+                    batchResponses.Add(body);
+                }
+
+                return Response.FromValue(
+                    new LogsBatchQueryResultCollection(batchResponses, batch),
+                    rawResponse);
+            }
+
+            using var message = _queryClient.CreateBatchRequest(new BatchRequest(batch.Requests));
+
+            TimeSpan? timeout = null;
+            foreach (var batchRequest in batch.Requests)
+            {
+                var requestTimeout  = batchRequest?.Options?.ServerTimeout;
+                if (requestTimeout != null &&
+                    (timeout == null || requestTimeout.Value > timeout.Value))
+                {
+                    timeout = requestTimeout;
+                }
+            }
+
+            if (timeout != null)
+            {
+                message.NetworkTimeout = timeout.Value.Add(_networkTimeoutOffset);
+                message.Request.Headers.SetValue("prefer", $"wait={(int) timeout.Value.TotalSeconds}");
```

Comment:
Can we add this common header to [HttpHeader](https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/core/Azure.Core/src/HttpHeader.cs)?

Reply:
I kinda hate that type :) I don't think it adds much value over a constant string tbh.